src/component/legend/LegendModel.ts

```diff
@@ -362,6 +362,13 @@ class LegendModel<Ops extends LegendOption = LegendOption> extends ComponentMode
                     name: dataItem as string
                 };
             }
+
+            const isNil = (val: any) => val === null || val === undefined;
+            // remove null legend names
+            if (isNil(dataItem) || isNil(dataItem.name)) {
```

Comment:
`isNil` can be removed and use `if (dataItem == null || dataItem.name == null)` instead.